<?xml version="1.0"?>
<launch>
    <!-- 启动降落平台检测器 -->
    <node name="landing_platform_detector" pkg="landing_detection" type="landing_detection.py" output="screen">
        <param name="min_circle_radius" value="30"/>
        <param name="max_circle_radius" value="200"/>
        <param name="white_threshold" value="200"/>
        <param name="cross_length_ratio" value="0.3"/>
    </node>
    
    <!-- 可选：启动简化版检测器 -->
    <!-- 
    <node name="simple_landing_detector" pkg="landing_detection" type="simple_landing_detector.py" output="screen">
        <param name="min_contour_area" value="1000"/>
        <param name="circularity_threshold" value="0.6"/>
    </node>
    -->
    
    <!-- 启动图像查看器来显示调试图像 -->
    <node name="debug_image_viewer" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/landing_platform/debug_image"/>
    </node>
    
</launch>
